tempel.el

```diff
@@ -348,10 +365,23 @@ PROMPT is the optional prompt/default value."
 
 (defun tempel--save ()
   "Save template file buffer."
-  (when-let (buf (get-file-buffer tempel-file))
-    (with-current-buffer buf
-      (when (and (buffer-modified-p) (y-or-n-p (format "Save file %s? " tempel-file)))
-        (save-buffer buf)))))
+  (let (modified)
+    (dolist (file (tempel--expand-locations))
+      (when-let (buff (get-file-buffer tempel-file))
+  	(push file modified)))
+    (when (or (and (= (length modified) 1)
+		   (y-or-n-p "Save template file %s? " (car modified)))
+	      (y-or-n-p "Save modified template files? ")))
+    (mapc #'save-buffer modified)))
+
+(defun tempel--template-files-modified-p ()
+  "Return non-nil if any template files have been modified since loading."
+  (cl-find-if (lambda (file)
+		(let ((mod (file-attribute-modification-time (file-attributes file))))
+		  (unless (equal tempel--modified mod)
```

Comment:
The `tempel--modified` check is not correct. Check the maximum value of the file modification time (of all files) against the variable `tempel--file-modified`. Furthermore `tempel--file-modified` should probably be called `tempel--paths-modified`.